lego/apps/events/views.py

```diff
@@ -271,6 +274,51 @@ def upcoming(self, request):
         serializer = self.get_serializer(queryset, many=True)
         return Response(serializer.data)
 
+    @decorators.action(
+        detail=True, methods=["GET"], serializer_class=EventAdministrateSerializer
+    )
+    def export(self, *args, **kwargs):
+        if not self.request.user.is_authenticated:
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+
+        has_permission = self.request.user.has_perm("administrate", obj=Event)
+        if not has_permission:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        query = (
+            Event.objects.filter(pk=kwargs["pk"])
+            .prefetch_related(
+                Prefetch(
+                    "pools__registrations",
+                    queryset=Registration.objects.select_related(
+                        "user"
+                    ).prefetch_related("user__abakus_groups"),
+                ),
+            )
+            .first()
+        )
+
+        event = self.get_serializer(query).data
+
+        response = HttpResponse(content_type="text/csv")
+        response[
+            "Content-Disposition"
+        ] = f'attachment; filename="{event["title"].replace(" ", "_")}_registrations.csv"'
+        writer = csv.writer(response)
+        writer.writerow(["username", "fullName", "emailAddress", "pool"])
+        for pool in event["pools"]:
```

Comment:
Should we maybe think about sending the csv to their email or something, so that we force people to avoid stupid things?

Reply:
It's not a bad idea :thinking:. I think this needs to be discussed further though.